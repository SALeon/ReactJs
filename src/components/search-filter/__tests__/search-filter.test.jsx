import React from 'react';
import { shallow, mount } from 'enzyme';
import SearchFilter from '../index';

const mockFilterItems = [{ id: '1', label: 'TITLE' }, { id: '2', label: 'GENRE' }];

describe('SearchFilter', () => {
  it('snapshot', () => {
    const wrapper = shallow(
      <SearchFilter
        searchConditionsItems={mockFilterItems}
        inputPlaceholder="enter search"
        searchLabel="SEARCH"
      />,
    );
    expect(wrapper).toMatchSnapshot();
  });

  describe('with enzyme mount', () => {
    let wrapper;
    beforeEach(() => {
      wrapper = mount(
        <SearchFilter
          searchConditionsItems={mockFilterItems}
          inputPlaceholder="enter search"
          searchLabel="SEARCH"
        />,
      );
    });

    it('check input', () => {
      const mockInput = 'data';
      wrapper.find('.search-field').simulate('change', { target: { value: mockInput } });
      expect(wrapper.state().inputValue).toBe(mockInput);
    });

    it('check submit', () => {
      wrapper.find('.filter__submit-button').simulate('click');
      // TODO: find out
      // wrapper.instance().handleSubmit();
      // expect(wrapper.instance().handleSubmit).toHaveBeenCalled();
    });
  });
});
