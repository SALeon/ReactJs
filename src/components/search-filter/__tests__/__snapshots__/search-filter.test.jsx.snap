// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchFilter snapshot 1`] = `
<div
  className="filter"
>
  <h2
    className="filter__name"
  >
    FIND YOUR MOVIE
  </h2>
  <SearchField
    classStyles="filter__input"
    inputHandler={[Function]}
    placeholder="enter search"
  />
  <h3
    className="filter__title"
  >
    SEARCH BY
  </h3>
  <ToggleList
    itemClassStyle="filter__item"
    itemContainerStyle="filter__item-container"
    items={
      Array [
        Object {
          "component": <button
            className="filter__item"
            type="button"
          >
            TITLE
          </button>,
          "id": "1",
        },
        Object {
          "component": <button
            className="filter__item"
            type="button"
          >
            GENRE
          </button>,
          "id": "2",
        },
      ]
    }
    listClassStyle="filter__list"
  />
  <Button
    classStyles="filter__submit-button"
    clickHandler={[Function]}
    label="SEARCH"
  />
</div>
`;
